app/Lib/Rscripts/analyse.R

```diff
@@ -2,25 +2,67 @@
+  # percentage correct and the item rest correlation (IRC) per item and
+  # per answer option and the cronbach's alpha for the whole exam.
+  #
+  # Args:
+  #   key: Matrix of 0's and 1's. key[i,j] implies wether answer option i
+  #        to item j is right (1) or wrong (0). If a column (item) consists of
+  #        only 0s, the item is interpreted as graded manually. This means
+  #        that the same column of input.answers is used directly as a score
+  #        (i.e. 1 is interpreted as a score of 1 instead of "answer option" 1).
+  #        Number of columns (items) should be at least 3 and equal to the
+  #        number of columns of input.answers and length of number.answeroptions.
+  #        There is no maximum number of columns or rows
+  #   input.answers: Ungraded matrix of answers. input.answers[i,j] is
+  #                  the answer of student (i) to item (j). Should consist of
+  #                  at least 3 columns (items) and 2 rows (students).
+  #                  Number of columns should be at least 3 and equal to the
+  #                  number of columns of key and length of number.answeroptions
+  #                  There is no maximum.
+  #   number.answersoptions: Vector with number of answer options per item.
+  #                          Length should be at least 3 and equal to number of
+  #                          columns of key and input.answers.
+  #                          There is no maximum length.
+  #
+  # Returns:
+  #  list with:
+  #   Cronbach's alpha
+  #   Maximum number of answer options
+  #   Vector of number of correct per item
+  #   Vector of percentage correct per item
+  #   Vector of IRC per item
+  #   Matrix[i,j] of number of students answering option i to item j
+  #     (only if any multiple choice items are present, else returns a 0)
+  #   Matrix[i,j] of percentage of students answering option i to item j
+  #     (only if any multiple choice items are present, else returns a 0)
+  #   Matrix[i,j] of IRC of answer option i to item j
+  #     (only if any multiple choice items are present, else returns a 0)
+
   number.students <- nrow(input.answers)
   number.questions <- ncol(input.answers)
 
   if (number.questions > 2 & number.students > 1) {
-    # Correct/Incorrect Matrix
-    input.correct <- matrix(0, number.students, number.questions) 
+    # Do the analysis only if there are at least 3 items and 2 students.
+
+    # Create Correct/Incorrect matrix
+    input.correct <- matrix(0, number.students, number.questions)
 
     # Fill in Correct/Incorrect Matrix
-    for (j in 1: number.questions) {
-      for (i in 1: number.students) {
+    for (j in 1 : number.questions) {
+      for (i in 1 : number.students) {
         if (!is.null(input.answers[i, j]) & all(key[, j] == 0)) {
           input.correct[i, j] <- input.answers[i, j]
-        } else if(any(input.answers[i, j] == which(key[, j] == 1))) {
-          input.correct[i, j] <- 1
+        } else {
```

Comment:
I don't think it is a problem to use the "else if" construction, because in that situation the else statement is still surrounded on the same line by curly braces.